### PR TITLE
Ability to share an OkHttpClient instance with Emitter.

### PR DESCRIPTION
**Pull request by R4md4c**
OkHttpClient documentation suggests to use a single instance across the whole application. 
So I thought it would be better to introduce an API that allows that, since it will allow sharing the ConnectionPool and ThreadPool of the client, therefore saving resources.

### CODE REVIEW

**Comment by snowplowcla** [comment]
Thanks for your pull request. Is this your first contribution to a Snowplow open source project? Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).

:memo: Please visit https://github.com/snowplow/snowplow/wiki/CLA to learn more and sign.

Once you've signed, please reply here (e.g. I signed it!) and we'll verify. Thanks.

**Comment by R4md4c** [comment]
I signed it!

**Comment by snowplowcla** [comment]
Confirmed! @R4md4c  has signed the Individual Contributor License Agreement. Thanks so much

File: snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/tracker/Emitter.java [hunk 1]
```diff
@@ -258,12 +270,19 @@ private Emitter(EmitterBuilder builder) {
...
+        }
+        else {
+            clientBuilder = builder.client.newBuilder();
+        }
+
+        client = clientBuilder.sslSocketFactory(tlsArguments.getSslSocketFactory(),
+                                                tlsArguments.getTrustManager())
+                              .connectTimeout(15, TimeUnit.SECONDS)
+                              .readTimeout(15, TimeUnit.SECONDS)
+                              .build();
```
Comment (BenFradet) [inline snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/tracker/Emitter.java]: I'm not really familiar with ok-http but if you already supplied a client, do you want to be recreating a builder? cc @ninjabear who has xp with ok-http

Comment (R4md4c) [reply snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/tracker/Emitter.java]: According to the documentation in [1]

> Customize your client with newBuilder()
> You can customize a shared OkHttpClient instance with newBuilder(). This builds a client that shares the same connection pool, thread pools, and configuration. Use the builder methods to configure the derived client for a specific purpose.
> 
> This example shows a call with a short 500 millisecond timeout:
   > OkHttpClient eagerClient = client.newBuilder()
       .readTimeout(500, TimeUnit.MILLISECONDS)
       .build();
  > Response response = eagerClient.newCall(request).execute();
 
So I'm just creating a new builder to share the internals of the supplied client and just customize the needed parts. 

[1] https://square.github.io/okhttp/3.x/okhttp/okhttp3/OkHttpClient.html

Comment (BenFradet) [reply snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/tracker/Emitter.java]: thanks for the explanation :+1:

**Comment by simlu** [comment]
We would love to see this merged in. Can someone comment on the status please?

**Comment by mhadam** [comment]
Hi everyone - we'll try to get this into the next release, which I don't have a firm timeline set but imagine we can get it within the next several weeks.

**Pull request closed without merging**